core/src/main/scala/kafka/log/Log.scala

```diff
@@ -68,8 +67,12 @@ case class LogAppendInfo(var firstOffset: Long,
                          targetCodec: CompressionCodec,
                          shallowCount: Int,
                          validBytes: Int,
-                         offsetsMonotonic: Boolean)
-
+                         offsetsMonotonic: Boolean,
+                         pid: Long,
+                         epoch: Short,
+                         firstSequence: Int,
+                         lastSequence: Int,
+                         consecutiveSequenceNumbers: Boolean)
```

Comment:
Might be clearer if I renamed it to `hasConsecutiveSequenceNumbers`. It's just a sanity check, though I guess it might be a bit better to raise an exception immediately instead of continuing with the other checks.